packages/compass-logging/src/index.ts

```diff
@@ -43,8 +46,30 @@ export function createLoggerAndTelemetry(component: string): {
   } as Writable;
   const writer = new MongoLogWriter('', null, target);
 
-  const track = (event: string, properties: Record<string, any> = {}): void => {
-    emit(ipc, 'compass:track', { event, properties });
+  const track = (...args: [string, TrackProps?]) => {
+    void Promise.resolve()
+      .then(() => trackAsync(...args))
+      .catch((error) => debug('track failed', error));
+  };
```

Comment:
Can we just make `track` an `async` function and wrap its body in `try/catch` to handle exceptions? I think that achieves the same thing but might look a bit cleaner. (Feel free to ignore this comment, I know you are trying to wrap the PR up :))

Reply:
This is my fault actually :). Everywhere we used `track` we had to make the call and the callback "fire and forget", and also remember to handle exceptions properly, this is a quick way to deal with all of those at once:

``` js
track('connected', async () => {
   await do_a_lot_of_very_slow_stuff_and_then_throw_5_errors();
   return {is_atlas: true}; 
});
```

Reply:
I’m not sure how that relates to simplifying the code here, though? I’m not suggesting a different API or different behavior, only to use a try/catch instead of wrapping everything in promises

Reply:
Ahh ok, sorry misunderstood that! won't you need `void track ..` and "remember" to not `await`  in the usage then? Not that it would change much... probably I'm also still missing something as usual :).

Reply:
Well … no need to `void` it unless the linter complains, but yeah, I guess that could be worked around by `const track = (...args: [string, TrackProps?]) => void trackAsync(...args)`.

Either way, I think it’s a bit odd to have a promise-based try/catch in a wrapper around an async function instead of just using a “real” try/catch inside the async function

Reply:
Ah ok, thanks! Now I understand, yeah could have `try/catch` inside the async function and here there would have been no need for it